docs/user_guide/notes.rst

```diff
@@ -0,0 +1,34 @@
+Notes for Administrators
+========================
+Make sure to completely read and understand the following before putting ``ngshare`` into production
+
+Race Condition
+--------------
+ngshare should NOT run concurrently, or there may be race conditions and data may be corrupted. For example, do not create multiple ngshare instances that share a same underlying database.
+
+Database Upgrade
+----------------
+ngshare checks database version every time it starts up. If database version is older than ngshare version, it performs schema and data migration. Normally this may only happen after ngshare is updated.
```

Comment:
```suggestion
ngshare checks the database version every time it starts up. If the database version is older than the ngshare version, it performs schema and data migration. Normally this only happens after ngshare is updated.
```

Reply:
Abigail and I changed it to "ngshare checks the database version every time it starts up. If the database version is older than the ngshare version, it performs schema and data migration. Normally, this may only happen after ngshare is updated."

I think "may" is necessary because sometimes migration does not happen after ngshare is updated.

Reply:
The current sentence has two possible meanings:
1. "Under normal circumstances, migration is only allowed after ngshare is updated. Migration can still occur against the rules. Behavior under atypical circumstances is not explained here." In this version, "may" refers to permission, as in "May I come in?".
2. "Under normal circumstances, migration only occurs after ngshare is updated or at some other time. Behavior under atypical circumstances is not explained here." In this version, "may" modifies the word "only", which leads to a contradiction.

None of these meanings seem to match what you are trying to say. I think there are two problems here. "Normally" isn't the right word to use, and "may" modifies the wrong thing. If I understand correctly, migration can only occur due to ngshare being upgraded, but not all upgrades cause a migration. 

Also, I think "this" technically refers to the migration, but some people may misread and think it refers to the entire process of checking the database version and performing a migration if necessary. My suggestion is to change the sentence to something like "Migrations happen only after ngshare is updated, but migrations do not necessarily occur after every update.", or simply "Migrations happen only after ngshare is updated." because the second part is not very important and is implied anyway.

Reply:
I think here it is better to be more explicit:

> ngshare checks the database version every time it starts up. If the database version is older than the ngshare version, it performs schema and data migration. 
> Under normal circumstances, migrations only happen after ngshare is updated and the update involves change in database structure. The ngshare database update log can be found in :doc:`/contributer_guide/alembic`.